pyairtable/api/types.py

```diff
@@ -25,6 +25,25 @@
 FieldName: TypeAlias = str
 
 
+class AITextDict(TypedDict, total=False):
+    """
+    A ``dict`` representing text generated by AI.
+
+    >>> record = table.get('recW8eG2x0ew1Af')
+    >>> record['fields']['Generated Text']
+    {
+        'state': 'generated',
+        'isStale': False,
+        'value': '...'
+    }
+    """
+
+    state: Required[str]
+    isStale: Required[bool]
+    value: Required[Optional[str]]
```

Comment:
The only type I'm not sure about is this - I assume this means string | null, but I'm not familiar with Python types 😄

Reply:
Recognizing the absurdity of having to write "required and optional", you've got it exactly right. If we enabled [postponed annotations](https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations), we could change this to `Required[str | None]`.